### PR TITLE
Commerce 1539 (6/7) Fix Serializable classes

### PR DESCRIPTION
**Pull request by alecsloan**

### CODE REVIEW

**Comment by alecsloan** [comment]
ci:test

**Comment by alecsloan** [comment]
ci:test

File: commerce-product-api/src/main/java/com/liferay/commerce/product/data/source/CPDataSourceResult.java [hunk 1]
```diff
@@ -77,6 +80,16 @@ public String toString() {
 		return sb.toString();
 	}
 
+	private void readObject(ObjectInputStream stream)
```
Comment (marco-leo) [inline commerce-product-api/src/main/java/com/liferay/commerce/product/data/source/CPDataSourceResult.java:83]: @alecsloan why?

Comment (alecsloan) [reply commerce-product-api/src/main/java/com/liferay/commerce/product/data/source/CPDataSourceResult.java:83]: If I remember correctly it complained about it being an implementation of Serializable but not having read or write methods.

**Comment by alecsloan** [comment]
@marco-leo Rebased. LMK if I need to change anything regarding your review.

**Comment by alecsloan** [comment]
ci:test

File: commerce-product-api/src/main/java/com/liferay/commerce/product/data/source/CPDataSourceResult.java (hunk 1 above)
Comment (ethanbustad) [reply commerce-product-api/src/main/java/com/liferay/commerce/product/data/source/CPDataSourceResult.java:83]: The javadoc for `Serializable` says:

> Classes that require special handling during the serialization and deserialization process must implement special methods with these exact signatures:

    private void writeObject(java.io.ObjectOutputStream out)
         throws IOException
    private void readObject(java.io.ObjectInputStream in)
         throws IOException, ClassNotFoundException;
    private void readObjectNoData()
         throws ObjectStreamException;

So I think we should only have to include these methods if we need custom de/serialization logic, which it looks like we don't.

**Comment by marco-leo** [comment]
won't merge as not really need

**Pull request closed without merging**